Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,122 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
-
-static FILE* get_stream(char const* filepath, char const* perms)
+static ErrorOr<void> write_line_content(StringView line, size_t count, bool show_duplicates, bool print_count, Core::Stream::File& outfile)
 {
-    FILE* ret;
-
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
+    if (!show_duplicates || count > 0) {
```

Comment:
I feel like this condition would be a lot easier to read if it were inverted:

```
if (show_only_duplicates && count <= 0)
    return {};

// Continue with printing...
```

Reply:
really? I find it pretty understandable: print if we don't want duplicates or that it's a duplicate

Reply:
I'm with tim here

Reply:
> I find it pretty understandable: print if we don't [only] want duplicates or that it's a duplicate

In text it's more understandable, but when reading the code some hours ago I needed quite a bit to wrap my head around that condition (although, that may have partially been caused by `count` starting from the "wrong" number).

On the other hand, "if we want only duplicates, and the current line isn't a duplicate, then don't print it" is still a lot more readable for me, both in text and code.